src/retriever/session.js

```diff
@@ -124,7 +123,7 @@ export class Session {
           REDIRECT_STATUS_CODES.includes(response.status)
         ) {
           const url =
-            that.redirectHistory[that.redirectHistory.length - 1].url.href; // response.config.url;
+            that.redirectHistory[that.redirectHistory.length - 1].url.href;
```

Comment:
Nit: This might be more readable.
```suggestion
            that.redirectHistory.at(-1).url.href;
```